core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,214 @@
+                $scope.SOLUTION_EDITOR_FOCUS_LABEL));
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.data = {
+              answerIsExclusive: false,
+              correctAnswer: null,
+              explanation: ''
+            };
+
+            $scope.submitAnswer = function(answer) {
+              $scope.data.correctAnswer = answer;
+            };
+
+            $scope.saveSolution = function() {
+              $modalInstance.close({
+                solution: SolutionObjectFactory.createNew(
+                  $scope.data.answerIsExclusive,
+                  $scope.data.correctAnswer,
+                  $scope.data.explanation)
+              });
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var correctAnswer = result.solution.correctAnswer;
+        var currentStateName = editorContextService.getActiveStateName();
+        var state = explorationStatesService.getState(currentStateName);
+        SolutionVerificationService.verifySolution(
+          explorationContextService.getExplorationId(),
+          state,
+          correctAnswer,
+          function () {
+            explorationStatesService.updateSolutionValidity(
+              currentStateName, true);
+            explorationWarningsService.updateWarnings();
+          },
+          function () {
+            explorationStatesService.updateSolutionValidity(
+              currentStateName, false);
+            explorationWarningsService.updateWarnings();
+            alertsService.addInfoMessage(INFO_MESSAGE_SOLUTION_IS_INVALID);
+          }
+        );
+
+        stateSolutionService.displayed = result.solution;
+        stateSolutionService.saveDisplayedValue();
+      });
+    };
+
+    $scope.deleteSolution = function(evt) {
+      evt.stopPropagation();
+
+      alertsService.clearWarnings();
+      $modal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/editor_tab/delete_solution_modal.html'),
+        backdrop: true,
+        controller: [
+          '$scope', '$modalInstance',
+          function($scope, $modalInstance) {
+            $scope.reallyDelete = function() {
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function() {
+        stateSolutionService.displayed = null;
+        stateSolutionService.saveDisplayedValue();
+        explorationStatesService.deleteSolutionValidity(
+          editorContextService.getActiveStateName());
+      });
+    };
+
+    $scope.onSaveInlineSolution = function() {
```

Comment:
Does this ever get called, now that we're only using modals for solution editing?